### ISSUE TITLE
[MPS] sum on a size=1 dim is ~5x slower than squeeze

### ISSUE THREAD

**Issue by janeyx99** [pytorch/pytorch#87129]
### 🐛 Describe the bug

For a dimension with size 1, MPS's implementation of sum is ~5x slower than squeeze. This becomes more painful when the tensor is larger, as shown in the benchmarks below.

## Proposed Solution:
Would it help to have MPS's kernel for sum call squeeze when the size of the sum'd dimension is already 1 (for the case where dimensions should not be kept)?

## Benchmarks
![image](https://user-images.githubusercontent.com/31798555/196279577-d624549b-ce8c-47d8-9058-df4702194020.png)
[sum_benchmark.txt](https://github.com/pytorch/pytorch/files/9804411/sum_benchmark.txt) --> changed this to a TXT so GH would let me upload it, but to run, change the extension back to .py.

### Versions

master
```
Collecting environment information...
PyTorch version: 1.14.0a0+gitb40f443
Is debug build: True
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.5 (arm64)
GCC version: Could not collect
Clang version: 14.0.0 (clang-1400.0.29.102)
CMake version: version 3.22.1
Libc version: N/A

Python version: 3.10.6 (main, Oct  7 2022, 15:17:36) [Clang 12.0.0 ] (64-bit runtime)
Python platform: macOS-12.5-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False

Versions of relevant libraries:
[pip3] numpy==1.23.1
[pip3] torch==1.14.0a0+gitb40f443
[conda] numpy                     1.23.1          py310h220015d_0  
[conda] numpy-base                1.23.1          py310h742c864_0  
[conda] torch                     1.14.0a0+gitb40f443           dev_0    <develop>
```

**Comment by zou3519** [comment]
torch.sum should always return a Tensor with new data (the schema for at::sum implies it is out-ofplace). Tensor.squeeze should always return a view. Because of that, we should expect the first case to almost always be slower, because it is returning a new Tensor with a copy of the original data instead of a view

**Comment by albanD** [comment]
Comparing it to clone should be more fair?

**Comment by DenisVieriu97** [comment]
In case of `squeeze` (and all other view ops), the data won't be materialized until some other operation tries to use that tensor (e.g an addition using that view tensor / printing it / clone). As @zou3519 mentioned, there won't be a copy during initialization of a view op. In case of sum, we'll actually have to perform the sum and return a new tensor. 
For CPU/CUDA there should be same problem.
E.g I am getting the following on CPU:
```
[------------------------------------------------------------------  ------------------------------------------------------------------]
                                                        |  torch.squeeze.max.item  |  torch.squeeze  |  torch.sum.max.item  |  torch.sum
1 threads: -----------------------------------------------------------------------------------------------------------------------------
      shape: torch.Size([16, 4096, 40, 1]) dim_size: 1  |         24016.8          |       6.3       |       200862.7       |   248915.3
      shape: torch.Size([16, 16, 40, 1]) dim_size: 1    |            53.1          |       6.2       |          455.1       |      427.5
```

**Comment by janeyx99** [comment]
Oh the fact that it's out of place does make sense that it'd be slower. However, adding in a clone() still shows a significant difference for big tensors.
![image](https://user-images.githubusercontent.com/31798555/196285620-26b0dd02-283b-40be-8f7e-ac8afa089413.png)

**Comment by janeyx99** [comment]
It could very much be valid that this same gap occurs on other envs, and if that is the case, I will remove the mps specific label.
